frontend/src/metabase/reference/metrics/MetricDetailContainer.jsx

```diff
@@ -63,15 +78,16 @@ export default class MetricDetailContainer extends Component {
   }
 
   render() {
-    const { isEditing, user, metric } = this.props;
+    const { location, user, metric } = this.props;
+    const isEditing = location.pathname.endsWith("/edit");
```

Comment:
This creates another place that needs to be kept in sync with `routes.jsx`. If I change the route there from "edit" to "editing", this will break. Instead, maybe pass `isEditing={true}` directly from the route?

e.g. something like this:

```js
<Route
  path="metrics/:metricId/edit"
  render={routeProps =>
    <MetricDetailContainer
      {...routeProps}
      isEditing={true}
    />
  }
/>
```

We still have the push/replace coupled to the route definition, but we should solve that separately by extracting templated route definitions.

Reply:
Good catch, thank you!

Reply:
@paulrosenzweig it appeared to be a bit trickier than I thought

As you can see from the [routes file](https://github.com/metabase/metabase/blob/ea4c591b88af7593508a73b8fa2b4d50d0edc0f9/frontend/src/metabase/routes.jsx#L5), we don't use the default react-routers `Route` component. There is a decorated version of it [here](https://github.com/metabase/metabase/blob/ea4c591b88af7593508a73b8fa2b4d50d0edc0f9/frontend/src/metabase/hoc/Title.jsx#L72) and it basically helps to change pages' meta `title` property.

So, the decorated Route takes a React element passed to it as a `component` prop. Then it wraps it with a HOC that manages meta titles. It does that by wrapping react-router's `Route` static property.

We need to be sure `routeProps` are passed to the final component, but we don't have access to them in Route's static method. Please, correct me if I'm wrong, I can miss something here

I'm worried if we want to make our `Route` support the `render` prop, we'll need to come up with some hacky solution that won't be easy to understand

What do you think if we just import react-router's `Route` here and alias it like `ReactRouterRoute`? It's the simplest thing I see, still, it will be a bit inconsistent and we won't be able to change the meta title for the Metric's reference page (but we don't do it anyway here)

Reply:
> I'm worried if we want to make our Route support the render prop, we'll need to come up with some hacky solution that won't be easy to understand

That makes sense to me. I think leaving the `isEditing` logic in the component itself is a good solution for now. If we're making similar changes elsewhere maybe we should reconsider and pull it up to the Route.